### ISSUE TITLE
ktlint format saves file with Unix line endings on Windows

### ISSUE THREAD

**Issue by thauk-copperleaf** [shyiko/ktlint#225]
I would expect either (A) the existing line endings to be preserved, or (B) the platform's native line endings to be used.

**Comment by shyiko** [comment]
Support for .editorconfig's end_of_line=crlf added in https://github.com/shyiko/ktlint/commit/2cd9546e2090a3876a15d8c9d7307db2e0fbd6ea (included in [0.24.0](https://github.com/shyiko/ktlint/releases/tag/0.24.0)).  
If crlf is used - ktlint tries to preserver it. If not - it fallback to .editorconfig's end_of_line (lf by default).

**Comment by thauk-copperleaf** [comment]
We check in `.editorconfig` where it is used by developers on Linux and Windows. Since we share the same file, we can't hard-code a single line ending type.

So unfortunately, the support you mention does not address the underlying issue.

**Comment by shyiko** [comment]
I'm not sure I understand.

`end_of_line=crlf` in `.editoconfig` affects both the editor (e.g. IntelliJ) and the ktlint. In other words, CRLF will be enforced regardless whether it's Linux, macOS or Windows (which is the whole point of having `end_of_line`).
On a separate note, I don't think I've ever seen a project using CRLF instead of LF (unless it's strictly Microsoft-stack-specific). 

If you have developers working on the same code both on Linux and Windows, using native line endings doesn't sound like a good idea (polluted git diffs, etc.) (unless you have a proper [.gitattributes](https://github.com/shyiko/ktlint/blob/master/.gitattributes)).

**Comment by thauk-copperleaf** [comment]
> In other words, CRLF will be enforced regardless whether it's Linux, macOS or Windows

But that's the problem. I want ktlint to use the native line endings of the platform it's running on. Otherwise when I run ktlint format, for one platform, the line endings of files will be changed.

As I said in my original description, I think ktlint should either (A) be configurable to use "platform native line endings", or (B) configurable to respect the existing line endings in the file when it's formatting it.

**Comment by shyiko** [comment]
Alright, I'm reopening the ticket as there seems to be an issue with CRLF not being preserved on Windows (case B) ([even though it should work](https://github.com/shyiko/ktlint/blob/master/ktlint-core/src/main/kotlin/com/github/shyiko/ktlint/core/KtLint.kt#L391)). For case A I'm going to add `end_of_line=native` support.
 
> A note on `autocrlf` - https://stackoverflow.com/a/2825829/482225 / https://stackoverflow.com/a/10855862/482225.  
TLDR: do not use crlf unless you absolutely have to. `.editoconfig`'s `end_of_line` + `.gitattributes` is all you need.

**Comment by shyiko** [comment]
`end_of_line=native` support added in [0.25.0](https://github.com/shyiko/ktlint/blob/master/CHANGELOG.md#0250---2018-07-25)
